manifests/exception.pp

```diff
@@ -108,23 +108,26 @@
         }
       }
       $fw_command = 'portopening'
-      validate_re($protocol,['^(TCP|UDP|ICMPv(4|6))$'])
-      if $protocol =~ /ICMPv(4|6)/ {
-        $allow_context = "protocol=${protocol}"
-      } else {
-        if empty($local_port) {
-          $local_port_cmd = ''
-        } else {
-          validate_re($local_port,['any|[0-9]{1,5}'])
-          $local_port_cmd = "${local_port_param}=${local_port}"
+      if empty($protocol) {
+        if(!empty($remote_port) or !empty($local_port)) {
+          fail 'Sorry, protocol is required, when defining local or remote port'
         }
-        if empty($remote_port) {
-          $remote_port_cmd = ''
+      } else {
+        if $protocol =~ /ICMPv(4|6)/ {
+          $allow_context = "protocol=${protocol}"
         } else {
-          validate_re($remote_port,['any|[0-9]{1,5}'])
-          $remote_port_cmd = " ${remote_port_param}=${remote_port}"
+          if empty($local_port) {
+            $local_port_cmd = ''
+          } else {
+            $local_port_cmd = "${local_port_param}=${local_port}"
+          }
+          if empty($remote_port) {
+            $remote_port_cmd = ''
+          } else {
+            $remote_port_cmd = " ${remote_port_param}=${remote_port}"
```

Comment:
stray space?

Reply:
Not really. This was already this way before. It's because of:
```
$allow_context = "protocol=${protocol} ${local_port_cmd}${remote_port_cmd}"
```
But, good catch, this is ugly; I will change it.

Reply:
Ah. I know why this was done that way. :-)
If remote_port_cmd isn't defined it will leave a trailing space at the end, which causes the test to fail. Two options. Either adapt the test, or the code. I'll go the latter route, since I believe we should not create commands with unnecessary trailing spaces (if they can be easily avoided).